src/main/java/core/basesyntax/service/OperationStrategyImpl.java

```
@@ -0,0 +1,19 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        return operationHandlerMap.get(operation);
+    }
```

Consider what happens if an operation is requested for which there is no handler in the map. This method will return `null`, which will likely cause a `NullPointerException` in the calling code. A more robust approach would be to check if the returned handler is null and throw an exception, for example, `UnsupportedOperationException`, to fail fast and provide a clear error message.